OneCommunityGlobal/HighestGoodNetworkApp
# Nahiyan_dark mode on Dashboard and Timelog page

# Description
This PR is only implementing dark mode on the Dashboard page and Timelog page.

## Related PRS (if any):
This frontend PR is related to the dev backend

## Main changes explained:
- Created a dark mode button on the nav bar
- New state variable
- Changes on dashboard components like Weekly Summaries, Summary Bar, Leaderboard, Tasks and Timelogs table, Badge section. 

## How to test:
1. check into the current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. Clear site data/cache
4. log as any user
5. go to the dashboard
6. verify each component flips to dark mode when the button clicked
7. Make sure they flip back to light mode
8. Do the same for the timelog page

## Screenshots or videos of changes:
![Screenshot 2024-04-01 174340](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/71025942/7ad7b0fd-90d5-4474-9606-78e5ce41a1b3)
![Screenshot 2024-04-01 174430](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/71025942/3368a027-1fbf-4433-b4b5-92fa4f978f9c)
![Screenshot 2024-04-01 174500](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/71025942/1e75e97d-9a1a-41fa-9774-ce999b589d56)
![Screenshot 2024-04-01 174529](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/71025942/492f1dcf-7e56-44c7-b4d3-0d7cd9970f59)
![Screenshot 2024-04-01 174556](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/71025942/49ebe4f2-e987-4959-a026-92877ffc92bd)


## Note:
If you have any suggestions to make dark mode better, please let me know.
